src/workers.ts

```diff
@@ -108,6 +108,7 @@ export class ThreadWorker<Data = any, Response = any> extends AsyncResource {
       .then(res => {
         this.parent?.postMessage({ data: res, id: value.id })
         this.lastTask = Date.now()
+        return res
```

Comment:
Could you explain why and if this return is needed? We don't return the result of the `fn` Promise chain anyways :thinking:

Reply:
It's the eslint plugin promise that enforces the rule promise/always-return, which a more than sensible rule that avoids stalled promise resolution in a promises call chain.
So returning the result is the most sensible way to silence that linter error.

Reply:
Ah! this is new, reported by `'promise'` eslint plugin
Okay :shrug:  
@pioardi What do you think about this rule? I will let you decide if we need this or not
IMO it isn't because the `res` will never be used somewhere and is just a pollution to nowhere
I really like technical clean code ^^'

Reply:
The code is correct if you return, it's not correct if not, semantically speaking. 

You can also return an empty function but you have to return or throw to be on the safe side.

Reply:
I have no concern with the current code

Reply:
I also don't have any problems with the current (master-branch) code
Also every tests are running successfully on the current code
I would prefer to not add this to the code, because it looks like we have forgotten to use the returned value in any way

Reply:
Okay I think at least returning `null` is a compromise